engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/NodeDependency.scala

```diff
@@ -21,13 +23,30 @@ case class Parameter(name: String,
                      typ: TypingResult,
                      runtimeClass: Class[_],
                      restriction: Option[ParameterRestriction] = None,
+                     editor: Option[ParameterEditor] = None,
                      additionalVariables: Map[String, TypingResult] = Map.empty,
                      branchParam: Boolean = false) extends NodeDependency {
 
   def isLazyParameter: Boolean = classOf[LazyParameter[_]].isAssignableFrom(runtimeClass)
 
 }
 
+@ConfiguredJsonCodec(encodeOnly = true) sealed trait ParameterEditor
+
+@JsonCodec(encodeOnly = true) case class RawParameterEditor() extends ParameterEditor
```

Comment:
You shouldn't use `case class` with empty list of parameters - you have a `case object` for this.

Reply:
Done